src/plugins/enzymes/src/FindEnzymesDialog.cpp

```diff
@@ -589,6 +640,55 @@ void FindEnzymesDialog::accept() {
     QDialog::accept();
 }
 
+void FindEnzymesDialog::sl_handleSupplierSelectionChange(QStringList checkedSuppliers) {
+    const auto enzymes = EnzymesSelectorWidget::getLoadedEnzymes();
+    QList<SEnzymeData> visibleEnzymes;
+    for (const auto& enzyme : qAsConst(enzymes)) {
+        for (const auto& supplier : qAsConst(enzyme->suppliers)) {
+            CHECK_CONTINUE(checkedSuppliers.contains(supplier));
+
+            visibleEnzymes.append(enzyme);
+            break;
+
+        }
+    }
+    enzSel->setEnzymesList(visibleEnzymes);
+    static const QString notDefinedTr = EnzymesIO::tr(EnzymesIO::NOT_DEFINED_SIGN);
+    auto value = checkedSuppliers.join(SUPPLIERS_LIST_SEPARATOR).replace(notDefinedTr, EnzymesIO::NOT_DEFINED_SIGN);
```

Comment:
This way you will replace any supplier with "Not defined" (a valid English phrase) inside. This is a bug. The code should be written to give bugs a 0 chance to happen. Replace "Not defined" with some UUID-like string to avoid overlap.